src/ethereum/prague/fork.py

```diff
@@ -964,7 +970,18 @@ def process_transaction(
 
     output = process_message_call(message, env)
 
+    # For EIP-7623 gas_used includes the intrinsic costs with the floor price
+    # for calldata.
     gas_used = tx.gas - output.gas_left
+
+    # EIP-7623 floor price (note: no EVM costs)
+
+    # Transactions with less gas used than the floor pay at the floor cost.
+    if gas_used < floor:
+        gas_used = floor
```

Comment:
there is a risk here that sender's balance is less than the floor, because we did not charge the floor as part of the intrinsic gas

Reply:
We check against gas limit at the start and user always have enough balance to pay for the gas limit

Reply:
right